test/integration/restricted-input.js

```diff
@@ -0,0 +1,745 @@
+      var input = $('#credit-card-number');
+
+      input.click();
+
+      this.sendKeys('41115');
+
+      expect(input.getValue()).to.equal('4111 5');
+
+      browser.keys('ArrowLeft');
+      browser.keys('ArrowLeft');
+      browser.keys('Backspace');
+
+      expect(input.getValue()).to.equal('4115 ');
+    });
+
+    it('backspacing the character after a gap should keep the cursor after the gap', function () {
+      var selection;
+      var input = $('#credit-card-number');
+
+      input.click();
+
+      this.sendKeys('411156');
+
+      expect(input.getValue()).to.equal('4111 56');
+
+      browser.keys('ArrowLeft');
+      browser.keys('Backspace');
+
+      expect(input.getValue()).to.equal('4111 6');
+
+      selection = this.getSelectionRange('credit-card-number');
+      expect(selection.start).to.equal(5);
+      expect(selection.end).to.equal(5);
+    });
+
+    it('backspacing the character before a gap should backspace the character and move the gap', function () {
+      var selection;
+      var input = $('#credit-card-number');
+
+      input.click();
+
+      this.sendKeys('411156');
+
+      expect(input.getValue()).to.equal('4111 56');
+
+      browser.keys('ArrowLeft');
+      browser.keys('ArrowLeft');
+      browser.keys('ArrowLeft');
+      browser.keys('Backspace');
+
+      expect(input.getValue()).to.equal('4115 6');
+
+      selection = this.getSelectionRange('credit-card-number');
+      expect(selection.start).to.equal(3);
+      expect(selection.end).to.equal(3);
+    });
+
+    it('field overwrites', function () {
+      var input = $('#credit-card-number');
+
+      input.click();
+
+      this.sendKeys('1111111111111111');
+
+      expect(input.getValue()).to.equal('1111 1111 1111 1111');
+
+      this.repeat(function () {
+        browser.keys('ArrowLeft');
+      }, 19);
+
+      this.sendKeys('2222222222222222');
+
+      expect(input.getValue()).to.equal('2222 2222 2222 2222');
+    });
+
+    it('can backspace a whole field', function () {
+      var input = $('#credit-card-number');
+
+      input.click();
+
+      this.sendKeys('1111111111111111');
+      expect(input.getValue()).to.equal('1111 1111 1111 1111');
+
+      this.repeat(function () {
+        browser.keys('Backspace');
+      }, 16);
+      expect(input.getValue()).to.equal('');
+    });
+
+    it('can backspace in the middle', function () {
+      var input = $('#credit-card-number');
+
+      input.click();
+
+      this.sendKeys('1234567890123456');
+
+      expect(input.getValue()).to.equal('1234 5678 9012 3456');
+
+      this.repeat(function () {
+        browser.keys('ArrowLeft');
+      }, 11);
+
+      browser.keys('Backspace');
+      expect(input.getValue()).to.equal('1234 5689 0123 456');
+    });
+
+    it('can delete after a gap', function () {
+      if (this.browserName() === 'FIREFOX') {
+        this.skip('Firefox driver has a bug where the delete does not work');
```

Comment:
I was just thinking this should be a little clearer for when someone comes along who didn't experience this discovery
```suggestion
        this.skip('Firefox driver has a bug where the delete key does not work');
```